clustering.go

```
@@ -0,0 +1,286 @@
+		if _, ok := keepIdx[i]; ok {
+			out = append(out, c)
+		}
+	}
+	return out
+}
+
+// float64Bits is a tiny re-export point that consolidates the math.
+// Float64bits dependency so other tests/files don't have to import
+// "math" just to compare floats. Left as a package-private helper for
+// now — only dedupeChars's key construction uses it.
+func float64Bits(f float64) uint64 { return math.Float64bits(f) }
```

golangci-lint (2.12.2)

[error] func float64Bits is unused

(unused)